src/apps/bm_soft_module/user_code/user_code.cpp

```diff
@@ -0,0 +1,74 @@
+#include "user_code.h"
+#include "bm_network.h"
+#include "bm_printf.h"
+#include "bm_pubsub.h"
```

Comment:
new file to look at. This is a modified serial payload example, with all the serial stuff removed and just the soft sensor sampling and printing.